L2SIVacuum/POUs/Functions/Valves/Interfaces/FB_VFS_2OO3Interface.TcPOU

```diff
@@ -0,0 +1,185 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_VFS_2OO3Interface" Id="{289d35ca-42e8-4943-8216-657a27c694cb}" SpecialFunc="None">
+    <Declaration><![CDATA[(*Used as soft IO mapping to create a psuedo valve to communicate over two task on the same PLC.*) 
+(*NOTE: All 3 gauge cards with fast TTL output need to be installed in the same gauge controller*)
+(*For FAST shutter control*)
+FUNCTION_BLOCK FB_VFS_2OO3Interface EXTENDS FB_Valve
+VAR_INPUT
+	(*Make sure all three CC gauge cards are in the same controller*)
+	IG1 : ST_VG; // The MKS422 Cold Cathode Data Structure for CC gauge 1
+	IG2 : ST_VG; // The MKS422 Cold Cathode Data Structure for CC gauge 2
+	IG3 : ST_VG; // The MKS422 Cold Cathode Data Structure for CC gauge 3
+	Veto_Valve : ST_VGC; // The VGC structure for the Veto Valve
+END_VAR
+VAR_OUTPUT
+	{attribute 'pytmc' := '
+    pv: ;
+    '}
+	iq_stValve : ST_VFS; (* All valve data and states will be in this struct*)
+	i_xVAC_FAULT_OK AT%I* : BOOL; (*Valve Vacuum OK, is set to False when there is Vacuum Fault*)
+END_VAR
+VAR
+	rtTriggerVetoed : R_TRIG;
+	rtTriggered : R_TRIG;
+	tonOvrd	: TON;
+	tOvrd : TIME := T#10s;
+	rtOK : R_TRIG;
+	tonDelOK : TON;
+	xOPN_OK : BOOL;
+	tDelOK	: TIME := T#60S;
+
+	(*outputs*)
+	q_xPRESS_OK AT%Q* : BOOL;
+	q_xOPN_SW AT%Q* : BOOL;
+	q_xCLS_SW AT%Q* : BOOL; (*external open signal e.g epics*)
+	q_xVAC_FAULT_Reset AT%Q* : BOOL; (*Valve Vacuum OK, is set to False when there is Vacuum Fault*)
+	q_xOverrideMode AT%Q* : BOOL; (*To be linked to global override bit. This Overrides Vacuum logic only, EPS, MPS and PMPS are still enforces*)
+	q_xOverrideOpen AT%Q* : BOOL;
+	
+	(*VETO Devices*)
+	q_xVetoValveOpenDO AT%Q* : BOOL;
+	q_xVetoValveClosed AT%Q* : BOOL;
+	(*inputs*)
+	i_xTrigger AT%I* : BOOL;
+	i_xVFS_Open AT%I* : BOOL;
+	i_xVFS_Closed AT%I* : BOOL;
+	{attribute 'pytmc' := '
+    pv: MPS_FAULT_OK
+	field: ZNAM MPS FAULT ;
+	field: ONAM MPS OK ;
+	io: i ;
+	'}
+	i_xMPS_OK AT%I* : BOOL;(*MPS Fault OK, is set when the Valve is Open and there is no trigger*)
+	i_eVFS_State AT%I* : E_VGC; (*Interface*)
+END_VAR]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[(*2oo3 logic*)
+q_xPRESS_OK := (IG1.xPRESS_OK AND IG2.xPRESS_OK) OR
```

Comment:
I think this should be three separate x press ok bits sent to the VFS FB and evaluated in pairs with their respective triggers.